std/algorithm/sorting.d

```diff
@@ -1225,26 +1221,37 @@ private template TimSortImpl(alias pred, R)
             stack[stackLen++] = Slice(i, runLen);
             i += runLen;
 
-            // Collapse stack so that (e1 >= e2 + e3 && e2 >= e3)
+            // Collapse stack so that (e1 > e2 + e3 && e2 > e3)
             // STACK is | ... e1 e2 e3 >
             while (stackLen > 1)
             {
-                immutable run3 = stackLen - 1;
-                immutable run2 = stackLen - 2;
-                immutable run1 = stackLen - 3;
-                if (stackLen >= 3 && stack[run1].length <= stack[run2].length + stack[run3].length)
+                immutable run1 = stackLen - 4;
+                immutable run2 = stackLen - 3;
+                immutable run3 = stackLen - 2;
+                immutable run4 = stackLen - 1;
+                
+                if (stackLen > 2 && stack[run2].length <= stack[run3].length + stack[run4].length || 
+                    stackLen > 3 && stack[run1].length <= stack[run3].length + stack[run2].length )
```

Comment:
This may just be my OCD talking, but wouldn't this be clearer:
```D
if ((stackLen > 2 && stack[run2].length <= stack[run3].length + stack[run4].length) || (stackLen > 3 && stack[run1].length <= stack[run3].length + stack[run2].length))
```
(with line breaks where required)

This is primarily because I dislike non-explicit mixes of `&&` and `||` operators.